src/core/geocms/geonode/qgsgeonoderequest.cpp

```diff
@@ -272,9 +273,14 @@ QList<QgsGeoNodeRequest::ServiceLayerDetail> QgsGeoNodeRequest::parseLayers( con
   qint16 minorVersion;
   if ( jsonVariantMap.contains( QStringLiteral( "geonode_version" ) ) )
   {
-    const QStringList geonodeVersionSplit = jsonVariantMap.value( QStringLiteral( "geonode_version" ) ).toString().split( '.' );
-    majorVersion = geonodeVersionSplit.at( 0 ).toInt();
-    minorVersion = geonodeVersionSplit.at( 1 ).toInt();
+    QRegularExpression re( "((\\d+)(\\.\\d+))" );
+    QRegularExpressionMatch match = re.match( jsonVariantMap.value( QStringLiteral( "geonode_version" ) ).toString() );
+    if ( match.hasMatch() )
+    {
+      const QStringList geonodeVersionSplit = match.captured( 0 ).split( '.' );
+      majorVersion = geonodeVersionSplit.at( 0 ).toInt();
+      minorVersion = geonodeVersionSplit.at( 1 ).toInt();
+    }
```

Comment:
Else complain?

Reply:
If I'm not wrong, this always will find match if there are at least major and minor numbers in the version string. But added `else` branch to be safe

Reply:
That's correct, it looks rather stable. I just thought that the original author also assumed that there's always the format "x.y" :)